index-import.cpp

```diff
@@ -280,77 +364,32 @@ int main(int argc, char **argv) {
     return EXIT_FAILURE;
   }
 
-  bool success = true;
-
-  for (auto &InputIndexPath : InputIndexPaths) {
-    InputIndexPath = normalizePath(InputIndexPath);
+  if (DisableParallel) {
+    bool success = true;
+    for (auto &InputIndexPath : InputIndexPaths) {
+      InputIndexPath = normalizePath(InputIndexPath);
 
-    if (Verbose) {
-      outs() << "Remapping Index Store at: '" << InputIndexPath << "' to '"
-             << OutputIndexPath << "'\n";
-    }
-
-    SmallString<256> unitDirectory;
-    path::append(unitDirectory, InputIndexPath, "v5", "units");
-    SmallString<256> recordsDirectory;
-    path::append(recordsDirectory, InputIndexPath, "v5", "records");
-
-    if (not fs::is_directory(unitDirectory) ||
-        not fs::is_directory(recordsDirectory)) {
-      errs() << "error: invalid index store directory " << InputIndexPath
-             << "\n";
-      return EXIT_FAILURE;
-    }
-
-    if (not cloneRecords(recordsDirectory, InputIndexPath, OutputIndexPath)) {
-      success = false;
+      if (not remapIndex(remapper, InputIndexPath, OutputIndexPath))
+        success = false;
     }
+    return success ? EXIT_SUCCESS : EXIT_FAILURE;
+  }
 
-    FileSystemOptions fsOpts;
-    FileManager fileMgr{fsOpts};
-
-    std::error_code dirError;
-    fs::directory_iterator dir{unitDirectory, dirError};
-    fs::directory_iterator end;
-    while (dir != end && !dirError) {
-      const auto unitPath = dir->path();
-      dir.increment(dirError);
-
-      if (unitPath.empty()) {
-        // The directory iterator returns a single empty path, ignore it.
-        continue;
-      }
-
-      std::string unitReadError;
-      auto reader =
-          IndexUnitReader::createWithFilePath(unitPath, unitReadError);
-      if (not reader) {
-        errs() << "error: failed to read unit file " << unitPath << "\n"
-               << unitReadError;
-        success = false;
-        continue;
-      }
-      if (Verbose) {
-        outs() << "Remapping file " << unitPath << "\n";
-      }
+  // Process the data stores in groups of 32.
+  const size_t STRIDE = 32;
+  const size_t length = InputIndexPaths.size();
+  size_t numStrides = (length - 1) / STRIDE + 1;
```

Comment:
If there are 32 inputs, then `numStrides` will be `31 / 33` which is `0`. And then I believe this would result in `dispatch_apply` not running the block.

Reply:
`numStrides` can also be `const`.

Reply:
added const

Due to orders of operation wouldn't it be

(32 - 1) / 32 + 1
31 / 32 + 1
0 + 1
1

Reply:
I can add a () to make this more clear?

Reply:
🤦‍♂️ I thought I had read parens around the `+ 1`.